packages/docusaurus/src/webpack/base.ts

```diff
@@ -81,6 +81,19 @@ export function createBaseConfig(
         cacheGroups: {
           // disable the built-in cacheGroups
           default: false,
+          vendor: {
+            test: /[\\/]node_modules[\\/]/,
+            priority: 30,
+            minSize: 250000,
+            name(module) {
+              const packageName = module.context.match(
+                /[\\/]node_modules[\\/](.*?)([\\/]|$)/,
```

Comment:
Mind adding a comment to explain what this regex does?

Reply:
// get the package name. E.g. node_modules/packageName/not/this/part

I will sleep soon. Will add this tmrw :)